### DISCUSSION TITLE
Thorny question about `TRANSACTION_TIMESTAMP()`

### DISCUSSION THREAD

**Discussion by leiffoged** [cockroachdb/cockroach#91263] [Q&A]
If I write data that involves nodes A, B, and C, and then run `TRANSACTION_TIMESTAMP()` on node D immediately after A, B, and C acknowledge my write, can I then take the `TRANSACTION_TIMESTAMP()` value  (from node D)  and make an `AS OF SYSTEM TIME` query using that timestamp back to nodes A, B, or C and be guaranteed to see the data I wrote?

Wondering if this works or clock differences might make this sometimes fail. And, if this doesn't work, is there a good way (when committing a transaction) where I can get a timestamp that can be guaranteed to be after the transaction?

**Comment by ajwerner** [comment]
This is a timely question. There's been some demand for this in the past. See https://github.com/cockroachdb/cockroach/issues/79591. I have a partial implementation in https://github.com/cockroachdb/cockroach/pull/80848 which has come again just this week. I intend to do something about it in the coming weeks. I hope to take the outcome of that and try to backport it into a 22.2 patch release later this winter.
